source/translated_table_model.ts

```diff
@@ -0,0 +1,188 @@
+      this.endTransaction();
+    }
+  }
+
+  private processSourceAdd(operation: AddRowOperation) {
+    const rowIndex = operation.index;
+    const row = []      
+    for(let column = 0; column < this.sourceTable.columnCount; column++) {
+      row.push(this.sourceTable.get(rowIndex, column));
+    }
+    this.translatedTable.push(row);
+    this.sourceRowIndices = this.sourceRowIndices.map(sourceIndex =>
+      sourceIndex >= rowIndex ? sourceIndex + 1 : sourceIndex);
+    this.sourceRowIndices.push(operation.index);
+    this.processOperation(new AddRowOperation(this.rowCount - 1));
+  }
+
+  private processSourceMove(operation: MoveRowOperation) {
+    const sourceIndex = operation.source;
+    const destinationIndex = operation.destination;
+    const translatedSourceIndex = this.sourceRowIndices.
+      findIndex(index => index === sourceIndex);
+    const translatedDestinationIndex = this.sourceRowIndices.
+      findIndex(index => index === destinationIndex);
+    this.translatedTable.move(translatedSourceIndex,
+      translatedDestinationIndex);
+    this.sourceRowIndices.splice(translatedDestinationIndex, 0,
+      this.sourceRowIndices.splice(translatedSourceIndex, 1)[0]);
+    this.sourceRowIndices = this.sourceRowIndices.map(translatedIndex => {
+      if(translatedIndex >= destinationIndex &&
+          translatedIndex < sourceIndex) {
+        return translatedIndex + 1;
+      } else if(translatedIndex > sourceIndex &&
+        translatedIndex <= destinationIndex) {
+      return translatedIndex - 1;
+      } else if(translatedIndex === sourceIndex) {
+        return destinationIndex;
+      } else {
+        return translatedIndex;
+      }
+    });
+    this.processOperation(new MoveRowOperation(translatedSourceIndex,
+      translatedDestinationIndex));
+  }
+
+  private processSourceRemove(operation: RemoveRowOperation) {
+    const rowIndex = operation.index;
+    const translatedIndex = this.sourceRowIndices.findIndex(index =>
+      index === rowIndex);
+    this.translatedTable.remove(translatedIndex);
+    this.sourceRowIndices.splice(translatedIndex, 1);
+    this.sourceRowIndices = this.sourceRowIndices.map((sourceIndex, index) =>
+      index >= rowIndex ? sourceIndex - 1 : sourceIndex);
+    this.processOperation(new RemoveRowOperation(translatedIndex));
+  }
+
+  private processSourceUpdate(operation: UpdateOperation) {
+    const rowIndex = operation.row;
+    const columnIndex = operation.column;
+    const value = this.sourceTable.get(rowIndex, columnIndex);
+    const translatedIndex = this.sourceRowIndices.findIndex(index =>
+      index === rowIndex);
+    this.translatedTable.set(translatedIndex, columnIndex, value);
+    this.processOperation(new UpdateOperation(translatedIndex, columnIndex));
+  }
+
+  private dispatcher: Kola.Dispatcher<Operation>;
+  private sourceTable: TableModel;
+  private sourceRowIndices: number[];
+  private translatedTable: ArrayTableModel;
```

Comment:
I don't think this is needed. TranslatedTable should be a wrapper on an existing table. You never have to worry about them not being in sync or not having the same data as the source table will always signal any changes.